Data-Structures-and-Algos-Topic/Module1-Intro-to-Data-Structures-and-Algos/Activity2_HashTables/cards/9.md

```diff
@@ -4,38 +4,83 @@
 
  **Example Output**:
 
- Input: "abcabcbb"
- Output: 3 
- Explanation: The answer is "abc", with the length of 3. 
+ Input: "ABDEFGABEF"
+ Output:  The input string is ABDEFGABEF The length of the longest non-repeating character substring is 6
+ Explanation: The answer is "BDEFGA" and “DEFGAB”, with the length of 6. 
 
  **Solution**:
 
  Concept: 
- The idea on how to approach this problem is to define the appropriate mapping of each character in your string based on it's index.
- This would skip repeated characters in the string to improve the run time of this algorithm. This would result of a time complexity of 
- O(n), a space complexity for the Hashmap of O(min(m,n)), and a space complexity for the table of O(m) * m.
+ The solution uses extra space to store the last indexes of already visited characters. The idea is to scan the string from left to right, keep track of the maximum length Non-Repeating Character Substring (NRCS) seen so far. Let the maximum length be max_len. When we traverse the string, we also keep track of the length of the current NRCS using cur_len variable. For every new character, we look for it in already processed part of the string (A temp array called visited[] is used for this purpose). If it is not present, then we increase the cur_len by 1. If present, then there are two cases:
```

Comment:
- typo; "indices"
- formatting: please use `when mentioning code snippets` (ex: `cur_len`). in fact all variable names, method calls, etc not in code blocks should be formatted in code snippets with the `(tick marks).